src/RouteCollection.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Router;
+
+use InvalidArgumentException;
+
+final class RouteCollection implements RouteCollectionInterface
+{
+    /**
+     * @var array
+     */
+    private array $items = [];
+
+    /**
+     * All attached routes as Route instances
+     *
+     * @var Route[]
+     */
+    private array $routes = [];
+
+    /**
+     * RouteCollection constructor.
+     * @param RouteCollectorInterface $collector
+     * @param string $prefix
+     * @param bool $buildRoutes
+     */
+    public function __construct(RouteCollectorInterface $collector)
+    {
+        $this->injectItems([$collector]);
```

Comment:
`injectItems()` expects `Route[]|Group[]` but we're passing `RouteCollectorInterface[]`. Either we need to update phpdoc or adjust code...